compressors/integer_coders.py

```diff
@@ -85,3 +86,149 @@ def test_universal_uint_encode_decode():
 
     is_lossless, _ = try_lossless_compression(data_block, encoder, decoder)
     assert is_lossless
+
+
+"""
+shared initialization for Golomb encoder and decoder
+"""
+
+
+def _initialize_golomb_coder(coder, M: int):
+    assert M > 0
+    coder.M = M
+    coder.b = int(math.floor(math.log2(coder.M)))
+    coder.cutoff = 2 ** (coder.b + 1) - coder.M
+    if coder.cutoff == coder.M:
+        coder.rice_code = True
+    else:
+        coder.rice_code = False
+
+
+class GolombUintEncoder(DataEncoder):
+    """
+    Golomb code with parameter M based on
+    https://en.wikipedia.org/wiki/Golomb_coding#Simple_algorithm:
+    If M is power of 2, we simplify to Rice codes with slight
+    change in logic.
+    """
+
+    def __init__(self, M: int):
+        _initialize_golomb_coder(self, M)
+
+    def encode_symbol(self, x: int):
```

Comment:
The VSCode add-on, should automatically add a very helpful docstring with the inputs/return etc. Do use that

Reply:
thanks, will do. If we could add the formatting expectations somewhere (Readme or in PR guidelines), that would be good. I just ran the black thing for now